src/KubernetesClient/Extensions.cs

```diff
@@ -0,0 +1,214 @@
+        /// <param name="result">The value of the object that was removed, or <see langword="null"/> if value was not present in dictionary</param>
+        /// <typeparam name="TKey">The type of key</typeparam>
+        /// <typeparam name="TValue">The type of value</typeparam>
+        /// <returns><see langword="true"/> if the object was removed from dictionry, or <see langword="false"/> if the specific key was not present in dictionary</returns>
+        public static bool Remove<TKey,TValue>(this IDictionary<TKey,TValue> source, TKey key, out TValue result)
+        {
+            result = default;
+            if (!source.TryGetValue(key, out result))
+                return false;
+            source.Remove(key);
+            return true;
+        }
+        /// <summary>
+        /// Tries to remove item from the queue
+        /// </summary>
+        /// <param name="queue">Source queue</param>
+        /// <param name="result">The result if dequeue was successful, other <see langword="null"/></param>
+        /// <typeparam name="T">The type of items in queue</typeparam>
+        /// <returns><see langword="true"/> if dequeue was successful, otherwise <see langward="false"/></returns>
+        public static bool TryDequeue<T>(this Queue<T> queue, out T result)
+        {
+            result = default;
+            if (queue.Count == 0)
+                return false;
+            try
+            {
+                result = queue.Dequeue();
+            }
+            catch (InvalidOperationException)
+            {
+                return false;
+            }
+            return true;
+        }
+        /// <summary>
+        /// Tries to look at the first item of the queue without removing it
+        /// </summary>
+        /// <param name="queue">The source queue</param>
+        /// <param name="result">The item at the top of the queue, or <see langword="null"/> if queue is empty</param>
+        /// <typeparam name="T">The type of the items in the queue</typeparam>
+        /// <returns><see langword="true"/> if the operation was successful, or <see langword="false"/> if the queue is empty</returns>
+        public static bool TryPeek<T>(this Queue<T> queue, out T result)
+        {
+            result = default;
+            if (queue.Count == 0)
+                return false;
+            try
+            {
+                result = queue.Peek();
+            }
+            catch (InvalidOperationException)
+            {
+                return false;
+            }
+
+            return true;
+        }
+        /// <summary>
+        /// Creates a <see cref="HashSet{T}"/> for <see cref="IEnumerable{T}"/>
+        /// </summary>
+        /// <param name="source">The source enumerable</param>
+        /// <typeparam name="T">The type of elements</typeparam>
+        /// <returns>The produced hashset</returns>
+        public static HashSet<T> ToHashSet<T>(this IEnumerable<T> source)
```

Comment:
Does this need to be a separate method?

Reply:
The method is used. It is present in .NET Core, but not in netstandard or net4x. It acts as polyfil.

Reply:
It would be preferable if all extensions methods are made internal except those which extend types declared by this project; otherwise, it simply pollutes the developer's completion list. `ToHashSet` being a case in point as it has been available since `netcoreapp2.0` and `net472`.